gluster-prometheus.spec

```diff
@@ -0,0 +1,126 @@
+
+%global gluster_prom_ver 1.0.0
+%global gluster_prom_rel rc0
+
+Name: %{repo}
+Version: %{gluster_prom_ver}
+Release: %{gluster_prom_rel}%{?dist}
+Summary: The GlusterFS prometheus metrics collectors
+License: GPLv2 or LGPLv3+
+URL: https://%{provider_prefix}
+%if 0%{?with_bundled}
+Source0: https://%{provider_prefix}/releases/download/v%{version}/%{name}-v%{gluster_prom_ver}-%{gluster_prom_rel}-vendor.tar.xz
+%else
+Source0: https://%{provider_prefix}/releases/download/v%{version}/%{name}-v%{gluster_prom_ver}-%{gluster_prom_rel}.tar.xz
+%endif
+
+ExclusiveArch: %{go_arches}
+
+BuildRequires: %{?go_compiler:compiler(go-compiler)}%{!?go_compiler:golang}
+BuildRequires: systemd
+
+%if ! 0%{?with_bundled}
+BuildRequires: golang(github.com/asaskevich/govalidator)
+BuildRequires: golang(github.com/cespare/xxhash)
+BuildRequires: golang(github.com/cockroachdb/cmux)
+BuildRequires: golang(github.com/coreos/etcd/clientv3)
+BuildRequires: golang(github.com/coreos/etcd/clientv3/concurrency)
+BuildRequires: golang(github.com/coreos/etcd/clientv3/namespace)
+BuildRequires: golang(github.com/coreos/etcd/embed)
+BuildRequires: golang(github.com/coreos/etcd/etcdserver/etcdserverpb)
+BuildRequires: golang(github.com/coreos/etcd/pkg/transport)
+BuildRequires: golang(github.com/coreos/etcd/pkg/types)
+BuildRequires: golang(github.com/coreos/pkg/capnslog)
+BuildRequires: golang(github.com/dgrijalva/jwt-go)
+BuildRequires: golang(github.com/golang/protobuf/proto)
+BuildRequires: golang(github.com/gorilla/handlers)
+BuildRequires: golang(github.com/gorilla/mux)
+BuildRequires: golang(github.com/justinas/alice)
+BuildRequires: golang(github.com/olekukonko/tablewriter)
+BuildRequires: golang(github.com/pborman/uuid)
+BuildRequires: golang(github.com/pelletier/go-toml)
+BuildRequires: golang(github.com/rasky/go-xdr/xdr2)
+BuildRequires: golang(github.com/sirupsen/logrus)
+BuildRequires: golang(github.com/spf13/cobra)
+BuildRequires: golang(github.com/spf13/pflag)
+BuildRequires: golang(github.com/spf13/viper)
+BuildRequires: golang(github.com/thejerf/suture)
+BuildRequires: golang(golang.org/x/net/context)
+BuildRequires: golang(golang.org/x/sys/unix)
+BuildRequires: golang(google.golang.org/grpc)
+BuildRequires: golang(go.opencensus.io)
+%endif
+
+Requires: glusterfs-server >= 5.0.0
+Requires: /usr/bin/strings
+%{?systemd_requires}
+
+%description
+The project gluster-prometheus provides set of metrics collectors which
+would be run on gluster storage nodes. These would generate metrics for
+consumption by prometheus server.
+
+%prep
+%setup -q -n %{name}-v%{gluster_prom_ver}-%{gluster_prom_rel}
+
+%build
+export GOPATH=$(pwd):%{gopath}
+mkdir -p src/%(dirname %{import_path})
+ln -s ../../../ src/%{import_path}
+
+pushd src/%{import_path}
+# Build gluster-prometheus
+%{gluster_prom_make} gluster-prometheus
+popd
+
+%install
+# Install gluster-prometheus
+%if "%{gluster_mgmt}" == "glusterd2"
```

Comment:
Do not customize here. Based on config it will support gd1/gd2

Reply:
We need to run `make install` with GLUSTER_MGMT variable for creation of systemd file. right?

Reply:
Once config changes merges, systemd unit file will use config file as parameter to start the service. So no build time customize required.

Reply:
Ack